src/main/java/rx/internal/operators/OnSubscribeSingle.java

```diff
@@ -0,0 +1,72 @@
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Single;
+import rx.SingleSubscriber;
+import rx.Subscriber;
+
+import java.util.NoSuchElementException;
+
+/**
+ * Allows conversion of an Observable to a Single ensuring that exactly one item is emitted - no more and no less.
+ * Also forwards errors as appropriate.
+ */
+public class OnSubscribeSingle<T> implements Single.OnSubscribe<T> {
+
+    private final Observable<T> observable;
+
+    public OnSubscribeSingle(Observable<T> observable) {
+        this.observable = observable;
+    }
+
+    @Override
+    public void call(final SingleSubscriber<? super T> singleSubscriber) {
+        observable.subscribe(new Subscriber<T>() {
```

Comment:
There is also a constructor overload that takes the child Subscriber to automatically chain together unsubscriptions.

Sent from my iPhone

> On Jun 24, 2015, at 3:54 AM, Lalit Maganti <notifications@github.com> wrote:
> 
> In src/main/java/rx/internal/operators/OnSubscribeSingle.java:
> 
> > +
> > +/**
> > + * Allows conversion of an Observable to a Single ensuring that exactly one item is emitted - no more and no less.
> > + * Also forwards errors as appropriate.
> > + */
> > +public class OnSubscribeSingle<T> implements Single.OnSubscribe<T> {
> > +
> > +    private final Observable<T> observable;
> > +
> > +    public OnSubscribeSingle(Observable<T> observable) {
> > +        this.observable = observable;
> > +    }
> > +
> > +    @Override
> > +    public void call(final SingleSubscriber<? super T> singleSubscriber) {
> > +        observable.subscribe(new Subscriber<T>() {
> Just to double check: does this mean I need to add a few checks like 
> if (singleSubscriber.isUnsubscribed()) {
> unsubscribe();
> return;
> }
> in the 4 subscriber methods?
> 
> —
> Reply to this email directly or view it on GitHub.
>